pkg/brokerapi/openservicebroker/open_service_broker_client.go

```diff
@@ -299,6 +299,76 @@ func (c *openServiceBrokerClient) DeleteServiceBinding(instanceID, bindingID, se
 
 }
 
+func (c *openServiceBrokerClient) PollServiceInstance(ID string, req *brokerapi.LastOperationRequest) (*brokerapi.LastOperationResponse, error) {
+	q, err := createPollParameters(req)
+	if err != nil {
+		glog.Errorf("Failed to create query parameters for poll last operation: %v", err)
+		return nil, err
+	}
+	url := fmt.Sprintf(pollingFormatString, c.url, ID, q)
+	pollReq := brokerapi.LastOperationRequest{}
+	resp, err := sendOSBRequest(c, http.MethodGet, url, pollReq)
+	if err != nil {
+		glog.Errorf("Failed to create new HTTP request: %v", err)
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, errStatusCode{statusCode: resp.StatusCode}
+	}
+
+	lo := brokerapi.LastOperationResponse{}
+	if err := util.ResponseBodyToObject(resp, &lo); err != nil {
+		return nil, err
+	}
+	return &lo, nil
+}
+
+// createPollParameters creates the query parameter string from the LastOperationRequest
+// According to the spec, ServiceID and PlanID should be included, so fail requests
+// without them as it indicates programming error on our part.
+func createPollParameters(req *brokerapi.LastOperationRequest) (string, error) {
+	if req.ServiceID == "" {
+		return "", fmt.Errorf("LastOperationRequest is missing service_id")
+	}
+	if req.PlanID == "" {
+		return "", fmt.Errorf("LastOperationRequest is missing plan_id")
+	}
+
```

Comment:
I thought service_id and plan_id were optional.

> Note: Although the request query parameters service_id and plan_id are not required, the platform should include them on all last_operation requests it makes to service brokers.

The above code would return an error because they are empty, and would not add the operation.